### PR TITLE
Allow accessing reviews for apps that aren't available in your region…

### PR DESCRIPTION
**Pull request by luckylavish**
… (BUG 1185614)

### CODE REVIEW

File: mkt/ratings/tests/test_views.py [hunk 1]
```diff
@@ -266,8 +266,7 @@ def test_filter_by_nonpublic_app(self, get_region_mock):
         get_region_mock.return_value = mkt.regions.USA
         res, data = self._get_filter(
             app=self.app.app_slug, expected_status=403)
-        eq_(data['detail'], 'The app requested is not public or not available '
-                            'in region "us".')
+        eq_(data['detail'], 'The app requested is not public "us".')
```
Comment (diox) [inline mkt/ratings/tests/test_views.py]: Remove the ` "us"`.

File: mkt/ratings/tests/test_views.py [hunk 2]
```diff
@@ -290,10 +289,10 @@ def test_filter_by_app_excluded_in_region(self, get_region_mock):
         AddonExcludedRegion.objects.create(addon=self.app,
                                            region=mkt.regions.BRA.id)
         get_region_mock.return_value = mkt.regions.BRA
```
Comment (diox) [inline mkt/ratings/tests/test_views.py]: You can remove this test method `test_filter_by_app_excluded_in_region` entirely.

File: mkt/ratings/views.py [hunk 3]
```diff
@@ -100,14 +100,13 @@ def get_app(self, ident):
         except Webapp.DoesNotExist:
             raise Http404
         current_region = get_region()
-        if ((not app.is_public() or
-             not app.listed_in(region=current_region)) and
+        if (not app.is_public() and
                 not check_addon_ownership(self.request, app)):
```
Comment (diox) [inline mkt/ratings/views.py]: can be reformated to:
```
        if not app.is_public() and not check_addon_ownership(
            self.request, app):
```

**Comment by diox** [comment]
Also, can you amend your commit message so that the "(bug xxxx)" part is part of the first line ? Thanks.

File: mkt/ratings/views.py (hunk 3 above)
Comment (luckylavish) [inline mkt/ratings/views.py]: Can't make it in one line, else flakes will raise an error as it will reach the words per line limit.

Comment (diox) [inline mkt/ratings/views.py]: yeah, I'm suggesting 2 lines by going to the next line after `check_addon_ownership(`

File: mkt/ratings/tests/test_views.py (hunk 1 above)
Comment (luckylavish) [inline mkt/ratings/tests/test_views.py]: Its the part of validation. As in line 266 its getting the value for region USA that is being compared with the error detail. If I will remove "us", test case will be failed.

File: mkt/ratings/views.py [hunk 4]
```diff
@@ -100,14 +100,13 @@ def get_app(self, ident):
...
             # App owners and admin can see the app even if it's not public
             # or not available in the current region. Regular users or
             # anonymous users can't.
             raise PermissionDenied(
-                'The app requested is not public or not available in region '
+                'The app requested is not public '
                 '"%s".' % current_region.slug)
```
Comment (diox) [inline mkt/ratings/views.py]: Change this to:
`raise PermissionDenied('The app requested is not public')`

(Removing the `'"%s".' % current_region.slug)` part)

Then you can make the change I suggest in test_views.py above.

Comment (luckylavish) [inline mkt/ratings/views.py]: but I think this is better, as it is giving complete information.

Comment (diox) [inline mkt/ratings/views.py]: The information is useless now that the region restriction has been removed from the view

File: mkt/ratings/views.py [hunk 5]
```diff
@@ -100,14 +100,13 @@ def get_app(self, ident):
         except Webapp.DoesNotExist:
             raise Http404
         current_region = get_region()
-        if ((not app.is_public() or
-             not app.listed_in(region=current_region)) and
-                not check_addon_ownership(self.request, app)):
+        if (not app.is_public() and not check_addon_ownership(
+                self.request, app)):
             # App owners and admin can see the app even if it's not public
             # or not available in the current region. Regular users or
```
Comment (diox) [inline mkt/ratings/views.py]: Please also update this comment to remove the part about the region

File: mkt/ratings/views.py [hunk 6]
```diff
@@ -100,14 +100,13 @@ def get_app(self, ident):
         except Webapp.DoesNotExist:
             raise Http404
         current_region = get_region()
-        if ((not app.is_public() or
-             not app.listed_in(region=current_region)) and
-                not check_addon_ownership(self.request, app)):
+        if (not app.is_public() and not check_addon_ownership(
```
Comment (diox) [inline mkt/ratings/views.py]: Now that the line is splitted after the `check_addon_ownership(` you should be able to remove the extra surrounding parenthesis

File: mkt/ratings/tests/test_views.py [hunk 7]
```diff
@@ -283,36 +280,20 @@ def test_filter_by_nonpublic_app_owner(self):
...
-                                           region=mkt.regions.BRA.id)
-        get_region_mock.return_value = mkt.regions.BRA
-        res, data = self._get_filter(
-            app=self.app.app_slug, expected_status=403)
-        eq_(data['detail'], 'The app requested is not public or not available '
-                            'in region "br".')
-
-    @patch('mkt.ratings.views.get_region')
-    def test_filter_by_app_excluded_in_region_admin(self, get_region_mock):
+    def test_filter_by_app_excluded_in_region_admin(self):
```
Comment (diox) [inline mkt/ratings/tests/test_views.py]: You can remove this test

File: mkt/ratings/tests/test_views.py [hunk 8]
```diff
@@ -283,36 +280,20 @@ def test_filter_by_nonpublic_app_owner(self):
...
                               rating=5)
         self.grant_permission(self.user, 'Apps:Edit')
         AddonExcludedRegion.objects.create(addon=self.app,
                                            region=mkt.regions.BRA.id)
-        get_region_mock.return_value = mkt.regions.BRA
         self._get_filter(app=self.app.app_slug)
 
-    @patch('mkt.ratings.views.get_region')
-    def test_filter_by_app_excluded_in_region_owner(self, get_region_mock):
+    def test_filter_by_app_excluded_in_region_owner(self):
```
Comment (diox) [inline mkt/ratings/tests/test_views.py]: You can remove this test as well

**Pull request merged**